synfig-studio/src/gui/trees/layertree.cpp

```diff
@@ -392,6 +397,39 @@ LayerTree::create_param_tree()
 	//column_time_track->set_visible(false);
 }
 
+bool
+LayerTree::search_param_tree(const Glib::RefPtr<Gtk::TreeModel>& model,int column,const Glib::ustring& search_string,const Gtk::TreeModel::iterator& iter)//make sure it is indeed search string
+{
+	Gtk::TreePath path(iter);
+	if(match_found && (path != match_path))// when match is found dont bother with others
+		return true;
```

Comment:
I don't see what this `if` and your state variables help.
I wrote:
```
bool
LayerTree::search_param_tree(const Glib::RefPtr<Gtk::TreeModel>& model,int column,const Glib::ustring& search_string,const Gtk::TreeModel::iterator& iter)//make sure it is indeed search string
{
	static int count1 = 0, count2 = 0;
	warning("b: %i a: %i", count1, count2);
	++count1;
	Gtk::TreePath path(iter);
	if(match_found && (path != match_path))// when match is found dont bother with others
		return true;
	++count2;
	error("\tb: %i a: %i", count1, count2);
```

And it always print both debug messages (warning and error) and the counters are always equal.

Reply:
this was to implement what was described in initial comment, but I think the functionality I was thinking of wasn't really necessary. Removed.